app/recurring_expenses/views.py

```diff
@@ -39,53 +40,102 @@ def create_or_edit_recurring_expense(request, expense_id=None):
         edit_recurring_expense(request, expense_id=expense_id)
 
 
+_IMPORT_CONTENT_KEYS = {
+    "json": "application/json",
+    "csv": "text/csv",
+}
+
+_IMPORT_CONTENT_TYPES = {value: key for key, value in _IMPORT_CONTENT_KEYS.items()}
+
+
+def load_json_data(file):
+    try:
+        return json.loads(file.read())
+    except json.JSONDecodeError:
+        raise ValueError("Invalid JSON data in the file")
+
+
+def load_csv_data(file):
+    try:
+        f = file.read().decode("utf-8")
+        return csv.reader(f.splitlines(), delimiter=",")
+    except csv.Error:
+        raise ValueError("Invalid CSV data in the file.")
+
+
+def upload_json_date(data: dict):
+    for item in data:
+        try:
+            recurring_expense = RecurringExpense(
+                id=item["id"] if "id" in item else None,
+                active=item["active"],
+                particulars=item["particulars"],
+                amount=item["amount"],
+                currency=item["currency"],
+                period=item["period"],
+            )
+            recurring_expense.save()
+        except KeyError:
+            raise ValueError("Missing fields in the JSON data")
+
+
+def upload_csv_data(reader):
+    next(reader, None)
+    rows_imported = 0
+    rows_errored = 0
+    total_rows = 0
+
+    for row in reader:
+        if row[0].strip(" ").lower() == "totals":
+            break
+
+        total_rows += 1
+
+        try:
+            recurring_expense = RecurringExpense(
+                id=UUID(row[0]) if row[0] else None,
```

Comment:
Not sure what the intended behavior here is? Unless the row is empty, there will always be a row[0] right?
If the row is completely empty, then we probably wont make it to this line

What are we trying to check here? Maybe it handles the following scenario

```
id,active,particulars,amount,currency,amount_nzd,period
00000000-0000-0000-0000-000000000000,True,Foo,1.00,NZD,1.00,1
,True,Bar,1.00,NZD,1.00,1
```

Reply:
I actually put that so that the ID in case the ID is not provided by the user in the import file, it's generated automatically, but we can have it mandatory as well.

Reply:
nah thats fine. I sort of realised the answer while I was typing the question I think its fine as it is?